docs/architecture/blockchain.md

```diff
@@ -0,0 +1,80 @@
+
+Third, it is possible to set an expiration window for transactions which result in the batch having an expiration window. If transaction `A` specifies it expires at block `X`, and transaction `B` specifies it expires at block `X-2`, but both end up in the same batch, the batch expiration will be set to the minimum of all transaction expirations.
+
+Forth, the set of notes resulting from all transactions in a batch must be free of duplicates. Which means that two valid transactions creating exactly the same note resulting in having the same nullifier and hence the note being consumable only once wouldn't be allowed.
+
+## Block production
+
+To create a `Block` multiple batches and their respective proofs are aggregated into a `Block`.
+
+`Block` production cannot happen in parallel and must be done by the Miden operator. In the future there will be several Miden operators competing for `Block` production. The schema used for `Block` production is similar to the one used in batch production - recursive verification. Multiple batch proofs are aggregated into a single `Block` proof. However, the `Block` also contains the commitments to the current global [state](state.md), the newly created nullifiers, the new state commitments for affected private accounts and the newly created notes, and the full states for all affected public accounts and newly created notes. The `Block` proof attests to the correct state transition from the previous `Block` root to the next and therefore to the change of the global state of Miden.
+
+<p style="text-align: center;">
+    <img src="../img/architecture/blockchain/block.png" style="width:90%;" alt="Account diagram"/>
+</p>
+
+> **Tip: Block contents**
+> - **State updates** only contain the hashes of updated elements. For example, for each updated account, we record a tuple `([account id], [new account hash])`.
+> - **ZK Proof** attests that, given a state commitment from the previous `Block`, there was a sequence of valid transactions executed that resulted in the new state commitment, and the output also included state updates.
+> - The `Block` also contains full account and note data for public accounts and notes. For example, if account `123` is an updated public account which, in the **state updates** section we'd see a record for it as `(123, 0x456..)`. The full new state of this account (which should hash to `0x456..`) would be included in a separate section.
+
+## Verifying blocks
+
+To verify that a `Block` corresponds to a valid (global) state transition, the following must be done:
+
+1. Compute hashes of public accounts and notes states.
+2. Ensure that these hashes match records in the **state updates** section (see picture).
+3. Verify the included `Block` proof against the following public inputs:
+   - State commitment from the previous `Block`.
+   - State commitment from the current `Block`.
+   - State updates from the current `Block`.
```

Comment:
I don't know yet what the public block inputs/output will look like exactly, but for now I would probably mention:
- Input: Previous block commitment
- Input: Commitment to a set of batches
- Output: Current block commitment

The current commitment is computed as part of proving the block, so it can't be an input to the block I think.
The state updates would then be covered by the commitment over the batches I believe.